test/e2e/daemon_set.go

```diff
@@ -427,12 +427,27 @@ func checkRunningOnAllNodes(f *framework.Framework, selector map[string]string)
 		framework.ExpectNoError(err)
 		nodeNames := make([]string, 0)
 		for _, node := range nodeList.Items {
+			if taintedNoSchedule(node) {
```

Comment:
I updated this part, but I didn't use `isNodeTainted`, because DaemonSet has its own predicates. I also added a unit test for the node condition "NetworkUnavailable" that DaemonSet controller accepts (we already tested "NotReady").